source/common/stats/symbol_table_impl.h

```diff
@@ -176,6 +181,9 @@ class SymbolTableImpl : public SymbolTable {
   // This must be held during both encode() and free().
   mutable Thread::MutexBasicLockable lock_;
 
+  // This must be held while updating stat_name_sets_.
+  mutable Thread::MutexBasicLockable stat_name_set_mutex_;
```

Comment:
Note we do need a separate mutex for this or we get deadlock detection triggered by absl when calling getDynamic().